src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -603,47 +603,64 @@ private async Task<string> CreatePullRequestAsync(List<UpdateAssetsParameters> u
-                            u => new SubscriptionPullRequestUpdate
-                            {
-                                SubscriptionId = u.update.SubscriptionId,
-                                BuildId = u.update.BuildId
-                            })
-                        .ToList()
-                };
+                    description.WriteLine("This pull request updates the following dependencies");
+                    description.WriteLine();
 
-                string prUrl = await darc.CreatePullRequestAsync(
-                    targetRepository,
-                    new PullRequest
+                    await CommitUpdatesAsync(requiredUpdates, description, darc, targetRepository, newBranchName);
+
+                    var inProgressPr = new InProgressPullRequest
                     {
-                        Title = await ComputePullRequestTitleAsync(inProgressPr),
-                        Description = description.ToString(),
-                        BaseBranch = targetBranch,
-                        HeadBranch = newBranchName
-                    });
+                        ContainedSubscriptions = requiredUpdates.Select(
+                                u => new SubscriptionPullRequestUpdate
+                                {
+                                    SubscriptionId = u.update.SubscriptionId,
+                                    BuildId = u.update.BuildId
+                                })
+                            .ToList()
+                    };
+
+                    string prUrl = await darc.CreatePullRequestAsync(
+                        targetRepository,
+                        new PullRequest
+                        {
+                            Title = await ComputePullRequestTitleAsync(inProgressPr),
+                            Description = description.ToString(),
+                            BaseBranch = targetBranch,
+                            HeadBranch = newBranchName
+                        });
 
-                inProgressPr.Url = prUrl;
+                    if (!string.IsNullOrEmpty(prUrl))
+                    {
+                        inProgressPr.Url = prUrl;
+
+                        await StateManager.SetStateAsync(PullRequest, inProgressPr);
+                        await StateManager.SaveStateAsync();
+                        await Reminders.TryRegisterReminderAsync(
+                            PullRequestCheck,
+                            null,
+                            TimeSpan.FromMinutes(5),
+                            TimeSpan.FromMinutes(5));
+                    }
+                    else
+                    {
+                        // Something wrong happened when trying to create the PR but didn't throw an exception (probably there was no diff). 
+                        // We need to delete the branch also in this case.
+                        await darc.DeleteBranchAsync(targetRepository, newBranchName);
```

Comment:
The callers appear to only check null, so you should probably return null here (or change callers) rather than accidentally return empty string.